wolfcrypt/src/asn.c

```diff
@@ -8680,18 +8806,23 @@ static int DecodeCertExtensions(DecodedCert* cert)
     if (input == NULL || sz == 0)
         return BAD_FUNC_ARG;
 
-    if (GetASNTag(input, &idx, &tag, sz) < 0) {
-        return ASN_PARSE_E;
-    }
+#ifdef WOLFSSL_CERT_REQ
+    if (!cert->isCSR)
+#endif
+    { /* Not included in CSR */
```

Comment:
But Context-Specific 0 should be!

Reply:
But a CRL doesn't use this API? Am I missing something here?

Reply:
Certificate Request not a CRL.
See https://tools.ietf.org/html/rfc5280

Reply:
Sorry. CSR not CRL.
Don't see extension in the RFC 2986 but do see attributes in a [0].

Reply:
This part is omitted on purpose in a CSR. A CSR cannot have extensions like a cert would but it can have an Extension Request attribute. [In this commit I added a CSR with an Extension Request attribute.](https://github.com/wolfSSL/wolfssl/pull/3439/commits/fc3fa66d5afdbfe4b709dc27447b41d1eefde4a6) The structure of this CSR is (note the lack of a [0]):
```
julek@julek-MS-7A38:~/Projects/libest/test/UT$ openssl asn1parse -in csr.ext.der -inform DER -i
    0:d=0  hl=4 l= 562 cons: SEQUENCE          
    4:d=1  hl=4 l= 411 cons:  SEQUENCE          
    8:d=2  hl=2 l=   1 prim:   INTEGER           :03
   11:d=2  hl=2 l=  20 cons:   SEQUENCE          
   13:d=3  hl=2 l=  18 cons:    SET               
   15:d=4  hl=2 l=  16 cons:     SEQUENCE          
   17:d=5  hl=2 l=   3 prim:      OBJECT            :commonName
   22:d=5  hl=2 l=   9 prim:      UTF8STRING        :localhost
   33:d=2  hl=3 l= 159 cons:   SEQUENCE          
   36:d=3  hl=2 l=  13 cons:    SEQUENCE          
   38:d=4  hl=2 l=   9 prim:     OBJECT            :rsaEncryption
   49:d=4  hl=2 l=   0 prim:     NULL              
   51:d=3  hl=3 l= 141 prim:    BIT STRING        
  195:d=2  hl=3 l= 221 cons:   cont [ 0 ]        
  198:d=3  hl=2 l=  31 cons:    SEQUENCE          
  200:d=4  hl=2 l=   9 prim:     OBJECT            :challengePassword
  211:d=4  hl=2 l=  18 cons:     SET               
  213:d=5  hl=2 l=  16 prim:      PRINTABLESTRING   :IGCu/xNL4/0/wOgo
  231:d=3  hl=3 l= 185 cons:    SEQUENCE          
  234:d=4  hl=2 l=   9 prim:     OBJECT            :Extension Request
  245:d=4  hl=3 l= 171 cons:     SET               
  248:d=5  hl=3 l= 168 cons:      SEQUENCE          
  251:d=6  hl=3 l= 149 cons:       SEQUENCE          
  254:d=7  hl=2 l=   3 prim:        OBJECT            :X509v3 Subject Alternative Name
  259:d=7  hl=3 l= 141 prim:        OCTET STRING      [HEX DUMP]:30818A8704AC12340287047F0000018210626F626361742E636973636F2E636F6D82106A696D626F622E636973636F2E636F6D82136C6F63616C686F73742E636973636F2E636F6D8704AC12340287047F0000018210626F626361742E636973636F2E636F6D82106A696D626F622E636973636F2E636F6D82136C6F63616C686F73742E636973636F2E636F6D
  403:d=6  hl=2 l=  14 cons:       SEQUENCE          
  405:d=7  hl=2 l=   3 prim:        OBJECT            :X509v3 Key Usage
  410:d=7  hl=2 l=   1 prim:        BOOLEAN           :255
  413:d=7  hl=2 l=   4 prim:        OCTET STRING      [HEX DUMP]:030205E0
  419:d=1  hl=2 l=  13 cons:  SEQUENCE          
  421:d=2  hl=2 l=   9 prim:   OBJECT            :sha256WithRSAEncryption
  432:d=2  hl=2 l=   0 prim:   NULL              
  434:d=1  hl=3 l= 129 prim:  BIT STRING        
```
CSR attributes are read in `ParseCertRelative` where there is a comment `/* Read attributes */`.